pymeasure/experiment/workers.py

```diff
@@ -76,8 +76,26 @@ def __init__(self, results, log_queue=None, log_level=logging.INFO, port=None):
         self.log_queue = log_queue
         self.log_level = log_level
 
+        global log
+        log = logging.getLogger()
+        log.setLevel(self.log_level)
+        # log.handlers = []  # Remove all other handlers
+        # log.addHandler(TopicQueueHandler(self.monitor_queue))
+        # log.addHandler(QueueHandler(self.log_queue))
+
         self.context = None
         self.publisher = None
+        if self.port is not None and zmq is not None:
+            try:
+                self.context = zmq.Context()
+                log.debug("Worker ZMQ Context: %r" % self.context)
+                self.publisher = self.context.socket(zmq.PUB)
+                self.publisher.bind('tcp://*:%d' % self.port)
+                log.info("Worker connected to tcp://*:%d" % self.port)
+                time.sleep(0.3)  # wait so that the socket will be ready before starting to emit messages
+            except Exception:
+                log.exception("couldn't connect to ZMQ context")
+                # FIXME: this happily continues even if connecting to the context fails!
```

Comment:
What about this remark? Is this a problem (it's not completely clear to me what happens if connecting fails)

Reply:
I wanted to postpone that, but indeed it's better to do it right away - I let this fall back to proper values (as if zmq were not installed)